#154240248 Cache the nutritional_info dictionary in NutritionPlans

Betty-Kebenei opened the pull request:
#### What does the PR do?
* This PR implements caching of nutrition information of the NutritionPlan model in the nutrition app. Previously, the overview would render slowly if a user had many plans since too many DB-quires were being fired just to calculate the total calories. Caching the nutrition info reduces the number of times the queries are fired and this speeds up the rendering of the overview.
* The PR also implements signals that deletes the nutrition information from the cache every time a NutritionPlan, Meal and MealItem is added, edited or deleted.

#### Description of the tasks to be completed:
* Add a method (nutritional_info) to NutritionPlan model in the nutrition app to cache the nutrition information by adding the get_nutritional_values method to the cache.
* Add a method(get_nutritional_info) to CacheKeyMapper model in cache module in utils.
* Update the overview template in nutrition/templates/plan to pick up the nutritional_info method instead of get_nutritional_info method in the NutritionPlan model.
* Add a signal module to the nutrition app that deletes the nutrition information from the cache every time a NutritionPlan, Meal and MealItem is added, edited or deleted.
* Add an apps module to pass the signals to the nutrition app.
* Initialize the configuration of the nutrition app in the __init__ module with the apps module.

#### What are the relevant pivotal tracker stories?
* #154240248

#### Screenshots showing outputs:

* Using 40 dummy plans for testing.

* Loading time of the overview template before adding nutrition_info to cache (At the bottom in red).

<img width="804" alt="screen shot 2018-01-18 at 7 45 06 am" src="https://user-images.githubusercontent.com/19909781/35083415-204f246a-fc30-11e7-9cf6-8695cc83596c.png">




* Code to that adds nutrition_info to cache (wger/nutrition/models.py).

<img width="654" alt="screen shot 2018-01-18 at 9 14 36 am" src="https://user-images.githubusercontent.com/19909781/35083429-299efa54-fc30-11e7-8b7f-2668cfd52a5e.png">




* Loading time of the overview template after adding nutrition_info to cache.

<img width="796" alt="screen shot 2018-01-18 at 7 47 32 am" src="https://user-images.githubusercontent.com/19909781/35083423-258fac1a-fc30-11e7-9404-13cbc14ae8e1.png">

Commit:
[Feature #154240246)] Add travis integration to slack

Commit:
[Feature #2142536] Implement a read-only endpoint to view exercise details.

lym committed:
[Chore #154240248] Add caching to nutrition plan

Previously if a user had many plans (approx. more than 25), the
overview rendered slowly as too many DB-queries were fired just
to calculate the total calories.

This commit adds caching of the nutrition information which
improves on the performance.

lym committed:
[Chore #154240248] Add caching to nutrition plan

Previously if a user had many plans (approx. more than 25), the
overview rendered slowly as too many DB-queries were fired just
to calculate the total calories.

This commit adds caching of the nutrition information which
improves on the performance.

lym committed:
[Chore #154240248] Add signals to reset the cache

Add signals that trigger resets of the nutrition info cache introduced
in the previous commit.

lym committed:
[Chore #154240248] Add signals to reset the cache

Add signals that trigger resets of the nutrition info cache introduced
in the previous commit.

Betty-Kebenei committed:
[Chore #* #154240248] Add comments.

Betty-Kebenei committed:
[Chore #* #154240248] Add comments.

Commit:
[Feature #154240246] Add settings to disable continous emails from travis

Commit:
[Feature #154240246] Add fix for travis syntax error

mbithenzomo commented:
Try to find a way to optimise the `triggered_cache_reset` functions in `signals.py`. Also, a nitpick: let's stick to single quotes for this project.

Commit:
[Feature #154240246]Add bower dependecy

Commit:
[Feature #154240246]Add .env file

Commit:
[Feature # 154240246]Add Slack key encryption

Commit:
[Feature #154240246]Add slackkey encryption to notifications

lym committed:
[Chore #154240248] Refactor code

- Commit to using single quotes in `wger/nutrition/apps.py` for
  consistency.
- Merge the similar methods in `wger/nutrition/signals.py` into one,
  because the *DRY* principle is cool.

lym committed:
[Chore #154240248] Refactor code

- Commit to using single quotes in `wger/nutrition/apps.py` for
  consistency.
- Merge the similar methods in `wger/nutrition/signals.py` into one,
  because the *DRY* principle is cool.

Commit:
[Feature #154240246]Add slack token  to notifications

Commit:
[Feature #154240246]Add key to travis.yml

lym committed:
Merge branch 'ch-cache-nutritional-info-154240248' of github.com:andela/wg-justice-league into ch-cache-nutritional-info-154240248

Commit:
[Feature #154240246]Add channel token to travis.yml

Commit:
[Feature #154240246]Add encyrpted token

Commit:
[Feature #154240246]Add fix for Tarvis build

Commit:
[Feature #154240246]Add install gulp to travis.yml

Commit:
[Feature #154240246]Add install gulp globally  to travis.yml

Commit:
[Feature #154240246]Add Travis badge

Commit:
[Feature #154240246]Add Travis badge in rst format

Commit:
Merge pull request #4 from andela/ft-slack-integration-TravisCl-154240246

[Feature #154240246] Slack integration with Travis CI

yeenab committed:
[Chore #2142536] Fix merge conflicts

Files package.json and package-lock.json were having merge conflicts

yeenab committed:
[Chore #2142536] Correct pep8 warnings

Correct pep8 to make travis build pass

Commit:
Merge pull request #3 from andela/ft-exercise-infos-endpoint-154240249

#154240249 Create an endpoint to show all infos for an exercise (read-only)

lym committed:
Merge branch 'ch-cache-nutritional-info-154240248' of github.com:andela/wg-justice-league into ch-cache-nutritional-info-154240248

lym committed:
[Chore] Get rid of pep8 warnings
